app/src/main/cpp/skyline/services/nifm/IGeneralService.cpp

```diff
@@ -18,13 +20,94 @@ namespace skyline::service::nifm {
         return {};
     }
 
+    Result IGeneralService::GetCurrentNetworkProfile(type::KSession &session, ipc::IpcRequest &request, ipc::IpcResponse &response) {
+        const SfNetworkProfileData networkProfileData{
```

Comment:
```suggestion
        static constexpr SfNetworkProfileData NetworkProfileData{
```

I don' t think hardcoding this is okay though, games could easily rely on the currentAddress being correct. Also ideally pull dns settings from android